From 59484f1813652ecbc515b44646c1dd9d99989b6a Mon Sep 17 00:00:00 2001
From: Wade Farnsworth <wade_farnsworth@mentor.com>
Date: Thu, 27 Oct 2011 09:46:29 -0500
Subject: [PATCH 1/3] Initialize dentry in __stp_call_mmap_callbacks_with_addr()

When compiling a tapset, gcc complains that dentry may be used uninitialized in
this function.  Set it to NULL to prevent this error.

Signed-off-by: Wade Farnsworth <wade_farnsworth@mentor.com>
---
 runtime/task_finder.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/runtime/task_finder.c b/runtime/task_finder.c
index 70d5ba1..6be9b07 100644
--- a/runtime/task_finder.c
+++ b/runtime/task_finder.c
@@ -678,7 +678,7 @@ __stp_call_mmap_callbacks_with_addr(struct stap_task_finder_target *tgt,
 	struct vm_area_struct *vma;
 	char *mmpath_buf = NULL;
 	char *mmpath = NULL;
-	struct dentry *dentry;
+	struct dentry *dentry = NULL;
 	unsigned long length = 0;
 	unsigned long offset = 0;
 	unsigned long vm_flags = 0;
-- 
1.7.0.4

